src/main/java/bio/terra/service/search/SearchService.java

```diff
@@ -45,12 +49,37 @@ private void createEmptyIndex(Snapshot snapshot) {
         } catch (IOException e) {
             throw new PdaoException("Error creating index", e);
         }
+        return indexName;
+    }
+
+    private void createIndexMapping(String indexName, List<Map<String, Object>> values) {
+        PutMappingRequest request = new PutMappingRequest(indexName);
+        Map<String, Object> properties = Map.of("values", values);
+        Map<String, Object> source = Map.of("properties", properties);
+        request.source(source);
+        try {
+            client.indices().putMapping(request, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            throw new PdaoException("Error creating index mapping", e);
+        }
+    }
+
+    private void addIndexData(String indexName, List<Map<String, Object>> values) {
+        values.forEach(v -> {
+            BulkRequest request = new BulkRequest();
+            request.add(new IndexRequest(indexName)
+                .id(v.get("uuid").toString())
+                .source(v, XContentType.JSON)
+            );
+        });
     }
 
     public SearchIndexModel indexSnapshot(Snapshot snapshot, SearchIndexRequest searchIndexRequest)
         throws InterruptedException {
-        createEmptyIndex(snapshot);
         List<Map<String, Object>> values = bigQueryPdao.getSnapshotTableData(snapshot, searchIndexRequest.getSql());
```

Comment:
Does this approach load all the data into memory? We probably don't need to do it now, but I think we'll need a streaming version of this too at some point.

Reply:
Yes, it does.

Reply:
I added a TODO for this.